x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingContext.java

```diff
@@ -177,7 +177,7 @@ public boolean isEmpty() {
         return anomalyDetectionTasks.isEmpty()
             && snapshotUpgradeTasks.isEmpty()
             && dataframeAnalyticsTasks.isEmpty()
-            && modelAssignments.values().stream().allMatch(assignment -> assignment.totalTargetAllocations() == 0);
+            && modelAssignments.values().stream().allMatch(assignment -> assignment.getTaskParams().getNumberOfAllocations() == 0);
```

Comment:
This code is so confusing... 

Anyway, `assignment.totalTargetAllocations()` is the number of allocations assigned to ML nodes, but if there are no nodes, there are no assigned allocations. `assignment.getTaskParams().getNumberOfAllocations()` is the number that's intended.

I've also added a test for this.